files/rsyslog/rsyslog.sh

```diff
@@ -7,25 +7,35 @@ export RSYSLOG_WORKDIRECTORY=${RSYSLOG_WORKDIRECTORY:-/var/lib/rsyslog.pod}
 export MERGE_JSON_LOG=${MERGE_JSON_LOG:-false}
 export UNDEFINED_DEBUG=${UNDEFINED_DEBUG:-false}
 # if false, mmexternal undefined_field won't be called.
-export USE_MMEXTERNAL=${USE_MMEXTERNAL:-true}
 # if true, mmnormalize skipempty is executed.
 # effective when USE_MMEXTERNAL is false.
-export SKIP_EMPTY=${SKIP_EMPTY:-false}
 export RSYSLOG_FILE_READ_FROM_TAIL=${RSYSLOG_FILE_READ_FROM_TAIL:-off}
 export RSYSLOG_JOURNAL_READ_FROM_TAIL=${RSYSLOG_JOURNAL_READ_FROM_TAIL:-off}
 
-# Create a config file for undefined fields from env vars
+CDM_USE_UNDEFINED=${CDM_USE_UNDEFINED:-false}
+CDM_KEEP_EMPTY_FIELDS=${CDM_KEEP_EMPTY_FIELDS:-""}
+CDM_UNDEFINED_TO_STRING=${CDM_UNDEFINED_TO_STRING:-false}
+CDM_UNDEFINED_DOT_REPLACE_CHAR=${CDM_UNDEFINED_DOT_REPLACE_CHAR:-"UNUSED"}
 undefined_conf="/var/lib/rsyslog.pod/undefined.json"
-echo "{\"CDM_USE_UNDEFINED\" : ${CDM_USE_UNDEFINED:-false}, \
+if [ "${CDM_USE_UNDEFINED}" = "false" -a "${CDM_KEEP_EMPTY_FIELDS}" = "" -a "${CDM_UNDEFINED_TO_STRING}" = "false" -a "${CDM_UNDEFINED_DOT_REPLACE_CHAR}" = "UNUSED" ] ; then
```

Comment:
Thinking about this some more - I think the intention here is to "do the right thing" if the user *has not* set either `SKIP_EMPTY` or `USE_MMEXTERNAL`.  That is, if I do `oc set env ds/rsyslog USE_MMEXTERNAL=true` that means I want to use mmexternal even if I did not set CDM_USE_UNDEFINED, CDM_UNDEFINED_TO_STRING, etc.  You could print a message like "INFO: USE_MMEXTERNAL was set but none of the UNDEFINED fields were set.  Consider using SKIP_EMPTY=true instead." to let the user know that they may have done something unintentional.
Likewise if the user sets `SKIP_EMPTY=true` but one of `CDM_USE_UNDEFINED`, etc. undefined parameters are set.  You could print a message like "INFO: SKIP_EMPTY was set but one or more of CDM_USE_UNDEFINED, etc. were set.  Consider unsetting SKIP_EMPTY and using USE_MMEXTERNAL=true instead."

Reply:
Ok.  So, we don't set SKIP_EMPTY and USE_MMEXTERNAL automatically in rsyslog.sh, but we respect the users' settings.  If their configuration is not enough, like missing CDM_USE_UNDEFINED with USE_MMEXTERNAL, we give an appropriate warning.

Now, when nothing is set, the default behaviour should be SKIP_EMPTY=true && USE_MMEXTERNAL=false, right?